### PR TITLE
SmartContractDecoderBase impl ABIDAI

### PR DESCRIPTION
**Pull request by fanyinghao**

### CODE REVIEW

**Comment by fanyinghao** [comment]
@zhiquan911 updated

**Comment by fanyinghao** [comment]
@zhiquan911 https://github.com/blocktree/eosio-adapter/commit/a20aa6cf7e450bc511d4ab9c0f5606d06ad800c6 use case for cache manager

File: openwallet/contract.go [hunk 1]
```diff
@@ -93,15 +94,26 @@ func (decoder *SmartContractDecoderBase) GetTokenBalanceByAddress(contract Smart
...
+	//@require
 	SetABIInfo(address string, abi ABIInfo) error
 }
 
 // ABIInfo abi model
 type ABIInfo struct {
-	address string
-	abi     string
+	Address string
+	ABI     interface{}
```
Comment (zhiquan911) [inline openwallet/contract.go]: 为什么不用string类型？

Comment (zhiquan911) [reply openwallet/contract.go]: ABI一般都是JSON

Comment (fanyinghao) [reply openwallet/contract.go]: 减少每次调用的类型转换 @zhiquan911  https://github.com/blocktree/eosio-adapter/commit/a20aa6cf7e450bc511d4ab9c0f5606d06ad800c6#diff-742dc3ed3b4b71d2722f62c4330e9459 这个例子

**Pull request merged**